shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJob.java

```diff
@@ -80,7 +80,9 @@ public void execute(final ShardingContext shardingContext) {
             return;
         }
         log.info("start RuleAlteredJobScheduler, jobId={}, shardingItem={}", jobId, shardingItem);
-        RuleAlteredJobScheduler jobScheduler = new RuleAlteredJobScheduler(jobContext);
+        RuleAlteredJobScheduler jobScheduler = new RuleAlteredJobScheduler(jobContext, jobContext.getJobProcessContext());
+        jobScheduler.getInventoryTasks().addAll(jobContext.getInventoryTasks());
+        jobScheduler.getIncrementalTasks().addAll(jobContext.getIncrementalTasks());
```

Comment:
It's better to use constructor to initialize inventoryTasks and incrementalTasks.